### PR TITLE
EF9365 : Fix the hardware vector engine.

### PR DESCRIPTION
**Pull request by jfdelnero**
(This fix the ARC Squale Basic instruction and probably many others stuffs)

### CODE REVIEW

File: src/devices/video/ef9365.cpp [hunk 1]
```diff
@@ -428,37 +428,37 @@ void ef9365_device::set_busy_flag(int period)
 //  get_x_reg: Get the X register value
 //-------------------------------------------------
 
-unsigned int ef9365_device::get_x_reg()
+unsigned short ef9365_device::get_x_reg()
```
Comment (startaq) [inline src/devices/video/ef9365.cpp]: Please use either uint16_t or u16.

File: src/devices/video/ef9365.cpp [hunk 2]
```diff
@@ -428,37 +428,37 @@ void ef9365_device::set_busy_flag(int period)
...
-	return (m_registers[EF936X_REG_X_MSB]<<8) | m_registers[EF936X_REG_X_LSB];
+	return ((m_registers[EF936X_REG_X_MSB] & 0x0F)<<8) | m_registers[EF936X_REG_X_LSB];
 }
 
 //-------------------------------------------------
 //  get_y_reg: Get the Y register value
 //-------------------------------------------------
 
-unsigned int ef9365_device::get_y_reg()
+unsigned short ef9365_device::get_y_reg()
```
Comment (startaq) [inline src/devices/video/ef9365.cpp]: Same here.

File: src/devices/video/ef9365.cpp [hunk 3]
```diff
@@ -428,37 +428,37 @@ void ef9365_device::set_busy_flag(int period)
...
-	return (m_registers[EF936X_REG_Y_MSB]<<8) | m_registers[EF936X_REG_Y_LSB];
+	return ((m_registers[EF936X_REG_Y_MSB] & 0x0F)<<8) | m_registers[EF936X_REG_Y_LSB];
 }
 
 //-------------------------------------------------
 //  set_x_reg: Set the X register value
 //-------------------------------------------------
 
-void ef9365_device::set_x_reg(unsigned int x)
+void ef9365_device::set_x_reg(unsigned short x)
```
Comment (startaq) [inline src/devices/video/ef9365.cpp]: And here and all the other places where you access hardware registers with a defined size.

File: src/devices/video/ef9365.cpp [hunk 4]
```diff
@@ -693,14 +701,14 @@ int ef9365_device::draw_vector(int x1,int y1,int x2,int y2)
...
-		}while( x <= x2 );
+		}while( x <= dest_x );
 	}
 	else
 	{
 		do
 		{
 			if(pen_state)
-				plot(x,y);
+				plot(x%bitplane_xres,y%bitplane_yres);
```
Comment (startaq) [inline src/devices/video/ef9365.cpp]: Please add some whitespace here to make it better readable.

File: src/devices/video/ef9365.cpp [hunk 5]
```diff
@@ -745,9 +753,12 @@ int ef9365_device::draw_vector(int x1,int y1,int x2,int y2)
 
 			x++;
 
-		}while( x <= x2 );
+		}while( x <= dest_x );
```
Comment (startaq) [inline src/devices/video/ef9365.cpp]: Spacing is weird here too. Our usual syntax would be:

`} while (x <= dest_x);`

File: src/devices/video/ef9365.cpp (hunk 1 above)
Comment (jfdelnero) [inline src/devices/video/ef9365.cpp]: Done.

File: src/devices/video/ef9365.cpp (hunk 2 above)
Comment (jfdelnero) [inline src/devices/video/ef9365.cpp]: Done.

File: src/devices/video/ef9365.cpp (hunk 3 above)
Comment (jfdelnero) [inline src/devices/video/ef9365.cpp]: Done.

File: src/devices/video/ef9365.cpp (hunk 4 above)
Comment (jfdelnero) [inline src/devices/video/ef9365.cpp]: Done.

Comment (jfdelnero) [inline src/devices/video/ef9365.cpp]: Done.

File: src/devices/video/ef9365.cpp (hunk 5 above)
Comment (jfdelnero) [inline src/devices/video/ef9365.cpp]: Done.

**Pull request merged**